nixos/modules/services/networking/v2raya.nix

```
@@ -2,15 +2,31 @@
 
 with lib;
 
+let
+  cfg = config.services.v2raya;
+in
+
 {
   options = {
     services.v2raya = {
       enable = options.mkEnableOption "the v2rayA service";
+
+      package = options.mkPackageOption pkgs "v2raya" { };
+      cliPackage = options.mkOption {
+        type = types.package;
+        default = pkgs.v2ray;
+        defaultText = literalExpression "pkgs.v2ray";
+        example = literalExpression "pkgs.xray";
+        description = ''
+          The package used for the cli binary.
+        '';
+        relatedPackages = [ "xray" "v2ray" ];
+      };
     };
   };
 
   config = mkIf config.services.v2raya.enable {
-    environment.systemPackages = [ pkgs.v2raya ];
+    environment.systemPackages = [ (pkgs.v2raya.override { v2ray = cfg.cliPackage; }) ];
```

```suggestion
    environment.systemPackages = [ (cfg.package.override { v2ray = cfg.cliPackage; }) ];
```
Ditto for the other instance of `pkgs.v2raya.override`.